### PR TITLE
Fixed bug on macOS, which caused the SP game to crash when reloading the map (after dying)

### PR DESCRIPTION
**Pull request by ram-on**
Refer to #154.  Credits go to @emileb.

### CODE REVIEW

File: SP/code/server/sv_game.c [hunk 1]
```diff
@@ -955,6 +955,13 @@ void SV_RestartGameProgs( void ) {
 	if ( !gvm ) {
 		Com_Error( ERR_FATAL, "VM_Restart on game failed" );
 	}
+	
+	// clear out old genitiy pointers so they are not accessed again
```
Comment (Mateos81) [inline SP/code/server/sv_game.c:959]: Little typo 'genitiy'  instead of 'gentity'

**Comment by emileb** [comment]
Thanks for setting this up. You might also need this: https://github.com/emileb/iortcw/commit/8321584a52babb03cf75e1021308cf31c45fca10

(Obviously without the Android stuff) I found it needed to return false (not use sever memory) otherwise it would still randomly crash on load (about 1 in 10 times)

**Comment by ram-on** [comment]
@emileb I haven't tested that commit (https://github.com/emileb/iortcw/commit/8321584a52babb03cf75e1021308cf31c45fca10), so I don't know if it works or not on macOS and whether there are any unwanted side effects.

I have tested this commit (https://github.com/iortcw/iortcw/pull/162) extensively and it resolved all crashes when loading maps.

**Comment by emileb** [comment]
> @emileb I haven't tested that commit ([emileb@8321584](https://github.com/emileb/iortcw/commit/8321584a52babb03cf75e1021308cf31c45fca10)), so I don't know if it works or not on macOS and whether there are any unwanted side effects.
> 
> I have tested this commit (#162) extensively and it resolved all crashes when loading maps.

Yep ok in that case makes sense to just go for this, maybe add the other fix later if needed.

**Comment by zturtleman** [comment]
According to a comment in Quake 3's SV_InitGameVM(), Quake 3 moved clearing client gentity pointers to before GAME_INIT to fix an issue. RTCW MP also does it before GAME_INIT but RTCW SP does it afterward.

It seems like this PR for RTCW SP would have the same affect as it clears the pointers right before calling SV_InitGameVM(). For consistency, it seem like using the Quake 3 / RTCW MP change in SV_InitGameVM() would be better. (I'm not really knowledgeable about this specific issue.)

SV_InitGameVM in [iortcw SP](https://github.com/iortcw/iortcw/blob/e92522e1a5bd0f1e220f005245e24f6abf469cd2/SP/code/server/sv_game.c#L931-L935)
```c
	// use the current msec count for a random seed
	// init for this gamestate
	VM_Call (gvm, GAME_INIT, sv.time, Com_Milliseconds(), restart);

	// clear all gentity pointers that might still be set from
	// a previous level
	for ( i = 0 ; i < sv_maxclients->integer ; i++ ) {
		svs.clients[i].gentity = NULL;
	}
```

SV_InitGameVM in [ioq3](https://github.com/ioquake/ioq3/blob/70d07d91d62dcdd2f2268d1ac401bfb697b4c991/code/server/sv_game.c#L881-L887) and [Quake 3](https://github.com/id-Software/Quake-III-Arena/blob/dbe4ddb10315479fc00086f08e25d968b4b43c49/code/server/sv_game.c#L899-L905). RTCW MP/[iortcw MP](https://github.com/iortcw/iortcw/blob/e92522e1a5bd0f1e220f005245e24f6abf469cd2/MP/code/server/sv_game.c#L921-L925) also has this but without the new comment.
```c
	// clear all gentity pointers that might still be set from
	// a previous level
	// https://zerowing.idsoftware.com/bugzilla/show_bug.cgi?id=522
	//   now done before GAME_INIT call
	for ( i = 0 ; i < sv_maxclients->integer ; i++ ) {
		svs.clients[i].gentity = NULL;
	}
	
	// use the current msec count for a random seed
	// init for this gamestate
	VM_Call (gvm, GAME_INIT, sv.time, Com_Milliseconds(), restart);
```

So my suggestion is to copy SV_InitGameVM() from iortcw MP to SP instead of the current change in this PR.

**Comment by MAN-AT-ARMS** [comment]
Looks like this was fixed in vanilla MP code at some point but wasn't changed in SP.
Merged into SP in 48116b99814d7c8350a25edf76de60e207053ca5

**Pull request closed without merging**